### DISCUSSION TITLE
Is there *any* way to let a shell script called in Helix know what the language is?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#6396] [Q&A]
```
[keys.normal.space]
c = ":sh kitty @ send-text --match 'title:^Terminal' cargo run \\\\n" 
t = ":sh kitty @ send-text --match 'title:^Terminal' 'clear \\ncargo test \\n'"
```
The above is perfect for Rust in Helix running in Kitty, but can we somehow send a shell script a string to say what the language is? Could the above be reworked to pipe a string containing the language ID into another script that returns the instruction to run and send it to the Kitty window? 

Maybe there is some totally different way to achieve this I can't think of? It would take Helix to the next level as the scripts could do so many different things!

EDIT: Maybe I answered my own question, https://github.com/helix-editor/helix/pull/3393 could be the perfect solution!

**Comment by Randalix** [comment]
What I use is something like:
kitty @ send-text --match 'title:^Terminal' '!! \r'
It will just run the last command. So in a rust project you would cargo run once manual, after that you can press the hotkey. I found this very handy, because it's very simple and universal.
